fuse/misc.go

```diff
@@ -23,9 +23,9 @@ func (code Status) String() string {
 			"NOTIFY_POLL",
 			"NOTIFY_INVAL_INODE",
```

Comment:
it never occurred to me that "inval" stands for invalidate. Maybe add a comment somewhere.

Reply:
Yes, FUSE protocol is not very explicitly documented, but it could be seen that `_INVAL_` indeed stands for invalidations:

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/linux/fuse.h?id=v4.19-rc6-177-gcec4de302c5f#n58
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/linux/fuse.h?id=v4.19-rc6-177-gcec4de302c5f#n389

Probably something as below would do:

```diff
--- a/fuse/types.go
+++ b/fuse/types.go
@@ -384,12 +384,12 @@ type NotifyStoreOut struct {
 }
 
 const (
-       //      NOTIFY_POLL         = -1
-       NOTIFY_INVAL_INODE = -2
-       NOTIFY_INVAL_ENTRY = -3
-       NOTIFY_STORE_CACHE = -4 // "notify store" in uapi/linux/fuse.h
-       //      NOTIFY_RETRIEVE_CACHE  = -5 // "notify retrieve" in uapi/linux/fuse.h
-       NOTIFY_DELETE      = -6
+       //      NOTIFY_POLL         = -1 // notify kernel that a poll waiting for IO on a file handle should wake up
+       NOTIFY_INVAL_INODE = -2 // notify kernel that an inode should be invalidated
+       NOTIFY_INVAL_ENTRY = -3 // notify kernel that a directory entry should be invalidated
+       NOTIFY_STORE_CACHE = -4 // store data into kernel cache of an inode
+       //      NOTIFY_RETRIEVE_CACHE  = -5 // retrieve data from kernel cache of an inode
+       NOTIFY_DELETE      = -6 // notify kernel that a directory entry has been deleted
 
 //     NOTIFY_CODE_MAX     = -6
 )
```

?

Also on this topic: I find that e.g. `fuse.Server.InodeNotify()` / / `nodefs.FileSystemConnector.FileNotify()` naming is a bit misleading. I would suggest to consider changing e.g. to `Server.NotifyInodeInvalidate()` and `FileSystemConnector.NotifyFileInvalidate()` but I'm not sure we can change public API.

( for the reference full names with such approach e.g. for FileSystemConnector:

- `NotifyFileInvalidate()`
- `NotifyEntryInvalidate()`
- `NotifyFileStoreCache()`
- `FileRetrieveCache()`   // not only it notifies, but also waits for data back from the kernel
- `NotifyEntryDelete()`

)

Reply:
we could change the name, but it seems a small improvement with a large cost.

If we do anything, there should probably be much more docs on how to use the library.

Reply:
I agree. Let's keep this in mind if we ever get to there.